diff -ur nx-2.0.0-o/nxviewer/nxviewer/cursor.c nx-2.0.0/nxviewer/nxviewer/cursor.c
--- nx-2.0.0-o/nxviewer/nxviewer/cursor.c	2006-01-09 06:43:52.000000000 -0700
+++ nx-2.0.0/nxviewer/nxviewer/cursor.c	2006-07-10 11:12:36.000000000 -0600
@@ -412,7 +412,6 @@
 	  offset = y * rcWidth + x;
 	  if (rcMask[offset]) {
 	    pos = (char *)&rcSource[offset * bytesPerPixel];
-	    //CopyDataToScreen(pos, x0, y0, 1, 1);
 	  }
 	}
       }
