diff -ur nx-2.0.0-o/nx-X11/lib/X11/Imakefile nx-2.0.0/nx-X11/lib/X11/Imakefile
--- nx-2.0.0-o/nx-X11/lib/X11/Imakefile	2006-09-08 17:56:29.000000000 -0600
+++ nx-2.0.0/nx-X11/lib/X11/Imakefile	2006-09-08 18:42:10.000000000 -0600
@@ -22,9 +22,9 @@
 #if NXLibraries
 
 #ifdef SunArchitecture
-NX_INCLUDES = -I../../../nxcomp -I/usr/sfw/include
+NX_INCLUDES = -I/usr/sfw/include
 #else
-NX_INCLUDES = -I../../../nxcomp
+NX_INCLUDES = 
 #endif
 
 NX_DEFINES = -DNX_TRANS_SOCKET
@@ -36,18 +36,8 @@
 #            -DNX_TRANS_TEST      \
 #            -DNX_TRANS_DEBUG     \
 
-NX_XCOMPCONFIGTARGET = ../../../nxcomp/config.status
-NX_XCOMPEXTCONFIGTARGET = ../../../nxcompext/config.status
 
-#ifdef cygwinArchitecture
-NX_XCOMPLIBTARGET = ../../../nxcomp/cygXcomp.dll
-NX_XCOMPEXTLIBTARGET = ../../../nxcompext/cygXcompext.dll
-#else
-NX_XCOMPLIBTARGET = ../../../nxcomp/libXcomp.so
-NX_XCOMPEXTLIBTARGET = ../../../nxcompext/libXcompext.so
-#endif
-
-NX_REQUIREDLIBS =  -L../../../nxcomp -lXcomp
+NX_REQUIREDLIBS =  -lXcomp
 
 #endif
 
@@ -1119,35 +1109,9 @@
 
 #if NXLibraries
 
-$(NX_XCOMPCONFIGTARGET):
-	cd ../../../nxcomp && \
-	./configure
-
-$(NX_XCOMPEXTCONFIGTARGET):
-	cd ../../../nxcompext && \
-	./configure
-
-#ifdef SunArchitecture
-$(NX_XCOMPLIBTARGET): $(NX_XCOMPCONFIGTARGET)
-	cd ../../../nxcomp && \
-	gmake
-
-$(NX_XCOMPEXTLIBTARGET): $(NX_XCOMPEXTCONFIGTARGET)
-	cd ../../../nxcompext && \
-	gmake
-#else
-$(NX_XCOMPLIBTARGET): $(NX_XCOMPCONFIGTARGET)
-	cd ../../../nxcomp && \
-	make
-
-$(NX_XCOMPEXTLIBTARGET): $(NX_XCOMPEXTCONFIGTARGET)
-	cd ../../../nxcompext && \
-	make
-#endif
-
-depend:: $(NX_XCOMPLIBTARGET) ks_tables.h
+depend:: ks_tables.h
 
-all::	 $(NX_XCOMPLIBTARGET) $(NX_XCOMPEXTLIBTARGET)
+all::	 
 
 #else
 
diff -ur nx-2.0.0-o/nx-X11/programs/Xserver/hw/nxagent/Imakefile nx-2.0.0/nx-X11/programs/Xserver/hw/nxagent/Imakefile
--- nx-2.0.0-o/nx-X11/programs/Xserver/hw/nxagent/Imakefile	2006-06-09 09:02:24.000000000 -0600
+++ nx-2.0.0/nx-X11/programs/Xserver/hw/nxagent/Imakefile	2006-09-08 18:41:43.000000000 -0600
@@ -130,14 +130,14 @@
 VFBINCLUDES = -I../../fb -I../../mfb -I../../render
 NXFONTINCLUDES = -I../../../../lib/font/include
 
-INCLUDES = -I. -I../../../../../nxcomp -I../../../../../nxcompext \
+INCLUDES = -I. \
 	   -I$(XBUILDINCDIR) -I$(FONTINCSRC) \
 	   -I../../mi -I../../include -I../../os \
 	   -I../../GL/glx -I../../GL/include -I../../../../lib/GL/include -I../../Xext \
            -I$(EXTINCSRC) -I$(XINCLUDESRC) \
 	   $(VFBINCLUDES) $(NXFONTINCLUDES)
 #ifdef SunArchitecture
-INCLUDES = -I../../../../../nxcomp -I../../../../../nxcompext \
+INCLUDES = \
 	   -I. -I$(XBUILDINCDIR) -I$(FONTINCSRC) \
 	   -I/usr/sfw/include \
 	   -I../../mi -I../../include -I../../os \
@@ -149,7 +149,7 @@
 INCLUDES = -I$(XBUILDINCDIR) -I$(FONTINCSRC) \
 	   -I../../mi -I../../include -I../../os \
 	   -I../../GL/glx -I../../GL/include -I../../../../lib/GL/include -I../../Xext \
-           -I. -I../../../../../nxcomp -I../../../../../nxcompext \
+           -I. \
            -I$(EXTINCSRC) -I$(XINCLUDESRC) \
 	   $(VFBINCLUDES) $(NXFONTINCLUDES)
 #endif
diff -ur nx-2.0.0-o/nx-X11/programs/Xserver/Imakefile nx-2.0.0/nx-X11/programs/Xserver/Imakefile
--- nx-2.0.0-o/nx-X11/programs/Xserver/Imakefile	2006-09-08 17:56:29.000000000 -0600
+++ nx-2.0.0/nx-X11/programs/Xserver/Imakefile	2006-09-08 18:46:41.000000000 -0600
@@ -1396,9 +1396,9 @@
  */
 
 #if defined(SunArchitecture)
-NXAGENTNXLIBS = -L ../../../nxcomp -L ../../../nxcompext -lXcomp -lXcompext -lrt -L/usr/sfw/lib -lXrender
+NXAGENTNXLIBS = -lXcomp -lXcompext -lrt -L/usr/sfw/lib -lXrender
 #else
-NXAGENTNXLIBS = -Wl,-rpath-link,../../../nxcomp -Wl,-rpath-link,../../../nxcompext
+NXAGENTNXLIBS = -lXcomp -lXcompext
 #endif
 
 #endif
@@ -1553,7 +1553,7 @@
 NXWINLIBS = PreFbLibs $(NXWINLIB) FbPostFbLibs $(LAYER) $(SHADOW) \
            $(XWINPARSERLIB)
 NXWINSYSLIBS = $(FONTLIBS) $(LDPRELIBS) $(XLIB) $(SYSLIBS) -lgdi32 -lddraw
-NXWINNXLIBS = -L../../../nxcomp -lXcomp
+NXWINNXLIBS = -lXcomp
 
 /*
  * These flags cause NXWin.exe to be a Windows executable, which
diff -ur nx-2.0.0-o/nx-X11/programs/Xserver/os/Imakefile nx-2.0.0/nx-X11/programs/Xserver/os/Imakefile
--- nx-2.0.0-o/nx-X11/programs/Xserver/os/Imakefile	2006-06-19 08:21:26.000000000 -0600
+++ nx-2.0.0/nx-X11/programs/Xserver/os/Imakefile	2006-09-08 19:27:38.000000000 -0600
@@ -25,7 +25,7 @@
 
 #if NXLibraries
 
-NX_INCLUDES = -I../../../../nxcomp
+NX_INCLUDES =
 
 NX_DEFINES = -DNX_TRANS_SOCKET        \
              -DNX_TRANS_AUTH          \
