diff -ur nx-2.0.0-o/nx-X11/config/cf/Imake.cf nx-2.0.0/nx-X11/config/cf/Imake.cf
--- nx-2.0.0-o/nx-X11/config/cf/Imake.cf	2006-06-19 08:18:55.000000000 -0600
+++ nx-2.0.0/nx-X11/config/cf/Imake.cf	2006-07-09 22:27:28.000000000 -0600
@@ -747,6 +747,13 @@
 #  define Mc68020Architecture
 #  undef mc68000
 # endif /* mc68000 */
+# ifdef __powerpc64__
+#  ifndef Ppc64Architecture
+#   define Ppc64Architecture
+#  endif
+#  undef __powerpc64__
+#  undef __powerpc__
+# endif
 # ifdef powerpc
 #  define PpcArchitecture
 #  undef powerpc
diff -ur nx-2.0.0-o/nx-X11/config/cf/linux.cf nx-2.0.0/nx-X11/config/cf/linux.cf
--- nx-2.0.0-o/nx-X11/config/cf/linux.cf	2006-06-19 08:18:55.000000000 -0600
+++ nx-2.0.0/nx-X11/config/cf/linux.cf	2006-07-09 22:27:28.000000000 -0600
@@ -288,6 +288,8 @@
 #  elif defined(HPArchitecture)
 #    define BuildXF86DRI	NO
 #  elif defined(i386Architecture)
+#  elif defined(Ppc64Architecture)
+#    define BuildXF86DRI	NO
 #    if OSMajorVersion > 2 || (OSMajorVersion == 2 && OSMinorVersion >= 2)
 #      define BuildXF86DRI	YES
 #    else
@@ -519,7 +521,7 @@
 #define MkdirHierCmd		mkdir -p
 
 #ifndef HaveLib64
-# if defined (x86_64Architecture) || defined (s390xArchitecture)
+# if defined (x86_64Architecture) || defined (s390xArchitecture) || defined (Ppc64Architecture)
 #  define HaveLib64	YES
 # else
 #  define HaveLib64	NO
@@ -809,6 +811,16 @@
 # define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #endif
 
+#ifdef Ppc64Architecture
+# define DefaultCCOptions	-fsigned-char -mminimal-toc
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	DefaultGcc2Ppc64Opt
+# endif
+# define LinuxMachineDefines	-D__powerpc64__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME -DPART_NET
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
+#endif /* Ppc64Architecture */
+
 #ifdef PpcArchitecture
 # define DefaultCCOptions	-fsigned-char
 # ifndef OptimizedCDebugFlags
diff -ur nx-2.0.0-o/nx-X11/config/cf/X11.tmpl nx-2.0.0/nx-X11/config/cf/X11.tmpl
--- nx-2.0.0-o/nx-X11/config/cf/X11.tmpl	2006-06-19 08:18:56.000000000 -0600
+++ nx-2.0.0/nx-X11/config/cf/X11.tmpl	2006-07-09 22:27:28.000000000 -0600
@@ -469,7 +469,7 @@
 #endif
 #endif
 #ifndef BuildXprintLib
-#define BuildXprintLib		(BuildXprint && !BuildServersOnly)
+#define BuildXprintLib		YES
 #endif
 #ifndef BuildScreenSaverExt
 #define BuildScreenSaverExt	NO
diff -ur nx-2.0.0-o/nx-X11/config/cf/xfree86.cf nx-2.0.0/nx-X11/config/cf/xfree86.cf
--- nx-2.0.0-o/nx-X11/config/cf/xfree86.cf	2006-06-19 08:18:56.000000000 -0600
+++ nx-2.0.0/nx-X11/config/cf/xfree86.cf	2006-07-09 22:27:28.000000000 -0600
@@ -1099,6 +1099,99 @@
 # endif
 #endif
 
+/*
+ * For PowerPC 64bit systems
+ * only libraries are strictly needed
+ */
+#if defined(Ppc64Architecture)
+# ifndef XF86Server
+#  define XF86Server		YES
+# endif
+/* Amiga framebuffer module */
+# ifndef XF86AFB
+#  define XF86AFB		NO
+# endif
+/* 1bpp module */
+# ifndef XF1Bpp
+#  define XF1Bpp		NO
+# endif
+/* 4bpp module */
+# ifndef XF4Bpp
+#  define XF4Bpp		NO
+# endif
+/* 8/32bpp overlay module */
+# ifndef XF8_32Bpp
+#  define XF8_32Bpp		YES
+# endif
+/* 8/16bpp dual fb module */
+# ifndef XF8_16Bpp
+#  define XF8_16Bpp		YES
+# endif
+/* 24/32bpp conversion module */
+# ifndef XF24_32Bpp
+#  define XF24_32Bpp		YES
+# endif
+/* shadow fb module */
+# ifndef XFShadowFB
+#  define XFShadowFB		YES
+# endif
+/* vgahw module */
+# ifndef XF86VgaHw
+#  define XF86VgaHw		NO
+# endif
+/* fbdevhw module */
+# ifndef XF86FBDevHw
+#  define XF86FBDevHw		YES
+# endif
+/* XAA module */
+# ifndef XF86XAA
+#  define XF86XAA		YES
+# endif
+/* ramdac module */
+# ifndef XF86Ramdac
+#  define XF86Ramdac		YES
+# endif
+/* I2C module */
+# ifndef XF86I2C
+#  define XF86I2C		YES
+# endif
+/* DDC module */
+# ifndef XF86DDC
+#  define XF86DDC		YES
+# endif
+/* RAC (Resource Access Control) module */
+# ifndef XF86RAC
+#  define XF86RAC		YES
+# endif
+
+/* int10 module */
+# ifndef XF86Int10
+#  define XF86Int10		YES
+# endif
+
+# ifndef DevelDrivers
+#  if XFree86Devel
+#   define DevelDrivers		
+#  else
+#   define DevelDrivers		/* */
+#  endif
+# endif
+
+/* Pure PCI drivers should go first */
+# ifndef XF86CardDrivers
+#  define XF86CardDrivers	ati mga \
+				DevelDrivers \
+				XF86OSCardDrivers XF86ExtraCardDrivers
+# endif
+# if HasGlide3
+#  define TdfxDriDriver		
+# else
+#  define TdfxDriDriver		/**/
+# endif
+# ifndef DriDrivers
+#  define DriDrivers		gamma r128
+# endif
+#endif
 
 /*
  * For Mips/Arc platforms, the default is to build all modules which
@@ -1970,6 +2063,12 @@
 # endif
 #endif
 
+#if HasGcc2 && defined(Ppc64Architecture)
+# ifndef DefaultGcc2Ppc64Opt
+#  define DefaultGcc2Ppc64Opt -O2 GccAliasingArgs
+# endif
+#endif
+
 #if HasGcc2 && defined(PpcArchitecture)
 # ifndef DefaultGcc2PpcOpt
 #  define DefaultGcc2PpcOpt -O2 GccAliasingArgs
diff -ur nx-2.0.0-o/nx-X11/config/imake/imakemdep.h nx-2.0.0/nx-X11/config/imake/imakemdep.h
--- nx-2.0.0-o/nx-X11/config/imake/imakemdep.h	2002-12-24 10:42:58.000000000 -0700
+++ nx-2.0.0/nx-X11/config/imake/imakemdep.h	2006-07-09 22:27:28.000000000 -0600
@@ -611,6 +611,9 @@
 # ifdef ppc
 	"-Dppc",
 # endif
+# ifdef ppc64
+	"-Dppc64",
+# endif
 # if defined(m68k)  || defined(M68k) || defined(m68040)
 	"-Dm68k",
 	"-DM68k",
@@ -670,6 +673,10 @@
 	"-DSVR4",
 # endif
 # ifdef __powerpc__
+#  ifdef __powerpc64__
+	"-D__powerpc64__",
+#  endif
+# else
 	"-D__powerpc__",
 # endif
 # ifdef PowerMAX_OS
@@ -704,6 +711,9 @@
 # ifdef __ppc__
         "-D__ppc__",
 # endif
+# ifdef __ppc64__
+        "-D__ppc64__",
+# endif
 # ifdef __i386__
         "-D__i386__",
 # endif
@@ -1227,6 +1237,10 @@
         {"__QNXNTO__", "1"},
 #endif
 # ifdef __powerpc__
+#  ifdef __powerpc64__
+	{"__powerpc64__", "1"},
+#  endif
+# else
 	{"__powerpc__", "1"},
 # endif
 # ifdef PowerMAX_OS
@@ -1301,6 +1315,9 @@
 #if defined(__ppc__)
         {"__ppc__", "1"},
 #endif
+#if defined(__ppc64__)
+        {"__ppc64__", "1"},
+#endif
 #if defined(__BIG_ENDIAN__)
       {"__BIG_ENDIAN__", "1"},
 #endif
diff -ur nx-2.0.0-o/nx-X11/include/Xmd.h nx-2.0.0/nx-X11/include/Xmd.h
--- nx-2.0.0-o/nx-X11/include/Xmd.h	2006-06-19 08:19:05.000000000 -0600
+++ nx-2.0.0/nx-X11/include/Xmd.h	2006-07-09 22:27:28.000000000 -0600
@@ -63,7 +63,8 @@
     defined(__sparc64__) || \
     defined(__s390x__) || \
     (defined(__hppa__) && defined(__LP64__)) || \
-    defined(__x86_64__) || defined(x86_64)
+    defined(__x86_64__) || defined(x86_64) || \
+    defined(__powerpc64__) || defined(powerpc64)
 #define LONG64				/* 32/64-bit architecture */
 #endif
 #ifdef __sgi
diff -ur nx-2.0.0-o/nx-X11/lib/X11/Imakefile nx-2.0.0/nx-X11/lib/X11/Imakefile
--- nx-2.0.0-o/nx-X11/lib/X11/Imakefile	2006-06-19 08:20:43.000000000 -0600
+++ nx-2.0.0/nx-X11/lib/X11/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -62,7 +62,7 @@
 #define DoExtraLib SharedLibX11
 #define DoDebugLib DebugLibX11
 #define DoProfileLib ProfileLibX11
-#define LibName X11
+#define LibName X11-nx
 #define SoRev SOXLIBREV
 #define HugeLibrary YES
 #define IncSubdir X11
Only in nx-2.0.0/nx-X11/lib/X11: Imakefile.orig
diff -ur nx-2.0.0-o/nx-X11/lib/X11/XlcDL.c nx-2.0.0/nx-X11/lib/X11/XlcDL.c
--- nx-2.0.0-o/nx-X11/lib/X11/XlcDL.c	2006-06-19 08:20:46.000000000 -0600
+++ nx-2.0.0/nx-X11/lib/X11/XlcDL.c	2006-07-09 22:27:28.000000000 -0600
@@ -58,10 +58,8 @@
 #ifdef _LP64
 # if defined(__sparcv9)
 #  define	_MACH64_NAME		"sparcv9"
-# elif defined(__ia64__) 
-#  undef MACH64_NAME
 # else
-#  error "Unknown architecture"
+#  undef MACH64_NAME
 # endif /* defined(__sparcv9) */
 # ifdef _MACH64_NAME
 #  define	_MACH64_NAME_LEN	(sizeof (_MACH64_NAME) - 1)
diff -ur nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/im/ximcp/Imakefile nx-2.0.0/nx-X11/lib/X11/xlibi18n/im/ximcp/Imakefile
--- nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/im/ximcp/Imakefile	2001-11-15 17:52:30.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/X11/xlibi18n/im/ximcp/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -13,7 +13,7 @@
 
    CONN_DEFINES = ConnectionFlags
    XTRANS_XIM_DEFINES = -DXIM_t -DTRANS_CLIENT
-   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11 -lc
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(imCallbk.c, ../../..)
 LinkSourceFile(imDefFlt.c, ../../..)
diff -ur nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/lc/def/Imakefile nx-2.0.0/nx-X11/lib/X11/xlibi18n/lc/def/Imakefile
--- nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/lc/def/Imakefile	2001-11-15 17:52:30.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/X11/xlibi18n/lc/def/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -4,7 +4,7 @@
       XI18NLIBNAME = xlcDef
            SRCS = lcDefConv.c
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11 -lc
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(lcDefConv.c, ../../..)
 
diff -ur nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/lc/gen/Imakefile nx-2.0.0/nx-X11/lib/X11/xlibi18n/lc/gen/Imakefile
--- nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/lc/gen/Imakefile	2001-11-15 17:52:30.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/X11/xlibi18n/lc/gen/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -4,7 +4,7 @@
       XI18NLIBNAME = xlibi18n
            SRCS = lcGenConv.c
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11 -lc
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(lcGenConv.c, ../../..)
 
diff -ur nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/lc/Utf8/Imakefile nx-2.0.0/nx-X11/lib/X11/xlibi18n/lc/Utf8/Imakefile
--- nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/lc/Utf8/Imakefile	2001-11-15 17:52:30.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/X11/xlibi18n/lc/Utf8/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -4,7 +4,7 @@
       XI18NLIBNAME = xlibi18n
            SRCS = lcUTF8Load.c
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11 -lc
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(lcUTF8Load.c, ../../..)
 
diff -ur nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/lc/xlocale/Imakefile nx-2.0.0/nx-X11/lib/X11/xlibi18n/lc/xlocale/Imakefile
--- nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/lc/xlocale/Imakefile	2001-11-15 17:52:30.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/X11/xlibi18n/lc/xlocale/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -4,7 +4,7 @@
       XI18NLIBNAME = xlocale
            SRCS = lcEuc.c lcSjis.c lcJis.c
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11 -lc
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(lcEuc.c, ../../..)
 LinkSourceFile(lcSjis.c, ../../..)
diff -ur nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/om/generic/Imakefile nx-2.0.0/nx-X11/lib/X11/xlibi18n/om/generic/Imakefile
--- nx-2.0.0-o/nx-X11/lib/X11/xlibi18n/om/generic/Imakefile	2001-11-15 17:52:30.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/X11/xlibi18n/om/generic/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -5,7 +5,7 @@
 		  omTextEsc.c omTextExt.c omTextPer.c omXChar.c
 
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11 -lc
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(omDefault.c, ../../..)
 LinkSourceFile(omGeneric.c, ../../..)
diff -ur nx-2.0.0-o/nx-X11/lib/Xaw/Imakefile nx-2.0.0/nx-X11/lib/Xaw/Imakefile
--- nx-2.0.0-o/nx-X11/lib/Xaw/Imakefile	2001-01-17 12:42:28.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/Xaw/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -194,6 +194,9 @@
 	XawI18n.o \
 	XawInit.o
 
+#undef DefaultCCOptions
+#define DefaultCCOptions GccWarningOptions
+
 #include <Library.tmpl>
 
 #if DoSharedLib && SharedDataSeparation
diff -ur nx-2.0.0-o/nx-X11/lib/Xaw6/Imakefile nx-2.0.0/nx-X11/lib/Xaw6/Imakefile
--- nx-2.0.0-o/nx-X11/lib/Xaw6/Imakefile	2002-05-31 12:45:44.000000000 -0600
+++ nx-2.0.0/nx-X11/lib/Xaw6/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -115,6 +115,9 @@
 
 INCLUDES = -I$(AWIDGETSRC)
 
+#undef DefaultCCOptions
+#define DefaultCCOptions GccWarningOptions
+
 #include <Library.tmpl>
 
 #if DoSharedLib && SharedDataSeparation
diff -ur nx-2.0.0-o/nx-X11/lib/Xext/Imakefile nx-2.0.0/nx-X11/lib/Xext/Imakefile
--- nx-2.0.0-o/nx-X11/lib/Xext/Imakefile	2003-05-06 07:29:38.000000000 -0600
+++ nx-2.0.0/nx-X11/lib/Xext/Imakefile	2006-07-09 22:30:25.000000000 -0600
@@ -10,14 +10,15 @@
 #define DoExtraLib SharedLibXext
 #define DoDebugLib DebugLibXext
 #define DoProfileLib ProfileLibXext
-#define LibName Xext
+#define LibName Xext-nx
 #define SoRev SOXEXTREV
 #define LibHeaders NO
 
 #include <Threads.tmpl>
 
 #ifdef SharedXextReqs
-REQUIREDLIBS = SharedXextReqs
+/* REQUIREDLIBS = SharedXextReqs $(NX_REQUIREDLIBS) */
+REQUIREDLIBS = -L../../exports/lib/ -lX11-nx $(NX_REQUIREDLIBS)
 #endif
 
 #if HasShm
Only in nx-2.0.0/nx-X11/lib/Xext: Imakefile~
Only in nx-2.0.0/nx-X11/lib/Xext: Imakefile.orig
Only in nx-2.0.0/nx-X11/lib/Xext: Imakefile.rej
Only in nx-2.0.0/nx-X11/lib/Xext: Imakefile.rej~
diff -ur nx-2.0.0-o/nx-X11/lib/Xrender/Imakefile nx-2.0.0/nx-X11/lib/Xrender/Imakefile
--- nx-2.0.0-o/nx-X11/lib/Xrender/Imakefile	2002-11-14 14:01:18.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/Xrender/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -74,7 +74,7 @@
 #define DoSharedLib SharedLibXrender
 #define DoDebugLib DebugLibXrender
 #define DoProfileLib ProfileLibXrender
-#define LibName Xrender
+#define LibName Xrender-nx
 #define SoRev SOXRENDERREV
 
 #ifdef XBuildIncDir
@@ -89,7 +89,8 @@
 #include <Threads.tmpl>
 
 #ifdef SharedXrenderReqs
-REQUIREDLIBS = SharedXrenderReqs
+/* REQUIREDLIBS = SharedXrenderReqs */
+REQUIREDLIBS = $(LDPRELIB) -L../../exports/lib/ -lXext-nx -lX11-nx
 #endif
 
 #if Malloc0ReturnsNull
diff -ur nx-2.0.0-o/nx-X11/lib/Xt/Xtos.h nx-2.0.0/nx-X11/lib/Xt/Xtos.h
--- nx-2.0.0-o/nx-X11/lib/Xt/Xtos.h	2002-04-04 07:05:36.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/Xt/Xtos.h	2006-07-09 22:27:28.000000000 -0600
@@ -66,7 +66,8 @@
     defined(__sparc64__) || \
     defined(__s390x__) || \
     (defined(__hppa__) && defined(__LP64__)) || \
-    defined(__x86_64__) || defined(x86_64)
+    defined(__x86_64__) || defined(x86_64) || \
+    defined(__powerpc64__) || defined(powerpc64)
 #define LONG64
 #endif
 
diff -ur nx-2.0.0-o/nx-X11/lib/Xxf86dga/Imakefile nx-2.0.0/nx-X11/lib/Xxf86dga/Imakefile
--- nx-2.0.0-o/nx-X11/lib/Xxf86dga/Imakefile	2003-01-03 19:48:26.000000000 -0700
+++ nx-2.0.0/nx-X11/lib/Xxf86dga/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -28,6 +28,9 @@
 
 #define IncludeSharedObjectInNormalLib
 
+#undef DefaultCCOptions
+#define DefaultCCOptions GccWarningOptions
+
 #include <Library.tmpl>
 
 DependTarget()
diff -ur nx-2.0.0-o/nx-X11/programs/xedit/lisp/Imakefile nx-2.0.0/nx-X11/programs/xedit/lisp/Imakefile
--- nx-2.0.0-o/nx-X11/programs/xedit/lisp/Imakefile	2002-12-13 21:41:14.000000000 -0700
+++ nx-2.0.0/nx-X11/programs/xedit/lisp/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -98,6 +98,9 @@
 		write.o \
 		xedit.o
 
+#undef DefaultCCOptions
+#define DefaultCCOptions GccWarningOptions
+
 #include <Library.tmpl>
 
 AllTarget(ProgramTargetName($(SAMP)))
diff -ur nx-2.0.0-o/nx-X11/programs/Xserver/fb/fb.h nx-2.0.0/nx-X11/programs/Xserver/fb/fb.h
--- nx-2.0.0-o/nx-X11/programs/Xserver/fb/fb.h	2003-01-30 14:46:30.000000000 -0700
+++ nx-2.0.0/nx-X11/programs/Xserver/fb/fb.h	2006-07-09 22:27:28.000000000 -0600
@@ -105,7 +105,8 @@
       defined(ia64) || defined(__ia64__) || \
       defined(__sparc64__) || \
       defined(__s390x__) || \
-      defined(x86_64) || defined (__x86_64__)
+      defined(x86_64) || defined (__x86_64__) || \
+      defined(powerpc64) || defined (__powerpc64__)
 typedef unsigned long	    FbBits;
 #  else
 typedef unsigned long long  FbBits;
diff -ur nx-2.0.0-o/nx-X11/programs/Xserver/hw/xfree86/fbdevhw/Imakefile nx-2.0.0/nx-X11/programs/Xserver/hw/xfree86/fbdevhw/Imakefile
--- nx-2.0.0-o/nx-X11/programs/Xserver/hw/xfree86/fbdevhw/Imakefile	2002-11-22 15:56:04.000000000 -0700
+++ nx-2.0.0/nx-X11/programs/Xserver/hw/xfree86/fbdevhw/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -1,6 +1,8 @@
 XCOMM $XFree86: xc/programs/Xserver/hw/xfree86/fbdevhw/Imakefile,v 1.12 2002/11/22 22:56:03 tsi Exp $
 
 #define IHaveModules
+#undef DefaultCCOptions
+#define DefaultCCOptions GccWarningOptions
 #include <Server.tmpl>
 
 #if defined(LinuxArchitecture) && LinuxFBDevSupport
diff -ur nx-2.0.0-o/nx-X11/programs/Xserver/hw/xfree86/os-support/bus/Imakefile nx-2.0.0/nx-X11/programs/Xserver/hw/xfree86/os-support/bus/Imakefile
--- nx-2.0.0-o/nx-X11/programs/Xserver/hw/xfree86/os-support/bus/Imakefile	2003-02-23 13:26:50.000000000 -0700
+++ nx-2.0.0/nx-X11/programs/Xserver/hw/xfree86/os-support/bus/Imakefile	2006-07-09 22:27:28.000000000 -0600
@@ -34,6 +34,7 @@
 
 #elif defined(LinuxArchitecture) && \
       (defined(PpcArchitecture) || \
+       defined(Ppc64Architecture) || \
        defined(MipsArchitecture) || \
        defined(ia64Architecture) || \
        defined(Mc68020Architecture) || \
diff -ur nx-2.0.0-o/nx-X11/programs/Xserver/hw/xfree86/os-support/bus/Pci.h nx-2.0.0/nx-X11/programs/Xserver/hw/xfree86/os-support/bus/Pci.h
--- nx-2.0.0-o/nx-X11/programs/Xserver/hw/xfree86/os-support/bus/Pci.h	2002-12-23 08:37:26.000000000 -0700
+++ nx-2.0.0/nx-X11/programs/Xserver/hw/xfree86/os-support/bus/Pci.h	2006-07-09 22:27:28.000000000 -0600
@@ -252,7 +252,7 @@
 #  define INCLUDE_XF86_MAP_PCI_MEM
 #  define INCLUDE_XF86_NO_DOMAIN
 # endif
-#elif defined(__powerpc__)
+#elif defined(__powerpc__) || defined(__powerpc64__)
 # if defined(linux)
 #  define ARCH_PCI_INIT linuxPciInit
 #  define INCLUDE_XF86_MAP_PCI_MEM
diff -ur nx-2.0.0-o/nx-X11/programs/Xserver/Imakefile nx-2.0.0/nx-X11/programs/Xserver/Imakefile
--- nx-2.0.0-o/nx-X11/programs/Xserver/Imakefile	2006-06-19 08:21:04.000000000 -0600
+++ nx-2.0.0/nx-X11/programs/Xserver/Imakefile	2006-07-09 22:29:33.000000000 -0600
@@ -1382,7 +1382,7 @@
 XPMLIB =  -lXpm
 NXAGENT = hw/nxagent/LibraryTargetName(nxagent)
 NXAGENTLIBS = PreFbLibs $(NXAGENT) FbPostFbLibs $(NXAGENT) $(MI)
-NXAGENTSYSLIBS = $(FONTLIBS) $(LDPRELIBS) $(XLIB) $(SYSLIBS) $(XPMLIB)
+NXAGENTSYSLIBS = $(FONTLIBS) $(LDPRELIBS) -lX11-nx -lXext-nx -lXrender-nx $(SYSLIBS) $(XPMLIB)
 #if HasParallelMake
 MakeMutex($(NXAGENTDIRS) $(NXAGENTOBJS) $(NXAGENTLIBS) $(NXAGENTSYSLIBS))
 #endif
@@ -1398,7 +1398,7 @@
 #if defined(SunArchitecture)
 NXAGENTNXLIBS = -L ../../../nxcomp -L ../../../nxcompext -lXcomp -lXcompext -lrt -L/usr/sfw/lib -lXrender
 #else
-NXAGENTNXLIBS = -L ../../../nxcomp -L ../../../nxcompext -lXcomp -lXcompext -lXrender -lX11 -lXext
+NXAGENTNXLIBS = -Wl,-rpath-link,../../../nxcomp -Wl,-rpath-link,../../../nxcompext
 #endif
 
 #endif
Only in nx-2.0.0/nx-X11/programs/Xserver: Imakefile~
Only in nx-2.0.0/nx-X11/programs/Xserver: Imakefile.orig
Only in nx-2.0.0/nx-X11/programs/Xserver: Imakefile.rej
diff -ur nx-2.0.0-o/nx-X11/programs/Xserver/os/xalloc.c nx-2.0.0/nx-X11/programs/Xserver/os/xalloc.c
--- nx-2.0.0-o/nx-X11/programs/Xserver/os/xalloc.c	2006-06-19 08:21:27.000000000 -0600
+++ nx-2.0.0/nx-X11/programs/Xserver/os/xalloc.c	2006-07-09 22:27:28.000000000 -0600
@@ -183,6 +183,7 @@
     defined(__ia64__) || defined(ia64) || \
     defined(__sparc64__) || \
     defined(__s390x__) || \
+    defined(__powerpc64__) || \
     defined(__x86_64__) || defined(x86_64)
 #define MAGIC			0x1404196414071968
 #define MAGIC_FREE              0x1506196615061966
