diff -ru nxviewer/nxviewer/Imakefile nxviewer.mod/nxviewer/Imakefile
--- nxviewer/nxviewer/Imakefile	2004-06-17 11:44:40.000000000 +0200
+++ nxviewer.mod/nxviewer/Imakefile	2004-09-14 15:33:31.000000000 +0200
@@ -51,6 +51,7 @@
 
 DEPLIBS = XawClientDepLibs $(VNCAUTH_LIB)
 
+#define XawClientLibs -L/usr/X11R6/lib -lXaw -lXmu -lXt -lSM -lICE -lXpm -lXext-nx -lX11-nx
 LOCAL_LIBRARIES =  XawClientLibs $(VNCAUTH_LIB) $(ZLIB_LIB) $(JPEG_LIB) $(NX_LIB)
 
 #ifdef SunArchitecture
